File: prdeploy-webhooks/src/models/settings/slack-settings.ts
import { EmailAliases } from '../email-aliases';
import { SlackWebhooksSettings } from './slack-webhooks-settings';

export interface SlackSettings {
  token: string;
  emailDomain: string;
  emailAliases: EmailAliases;
  webhooks: SlackWebhooksSettings;
  notificationsEnabled: boolean;
}

File: prdeploy-api/src/PrDeploy.Api.Models/Settings/SlackSettings.cs
using HotChocolate.Types;
using HotChocolate;

namespace PrDeploy.Api.Models.Settings;
public class SlackSettings
{
    public string? Token { get; set; }
    public string? EmailDomain { get; set; }
    [GraphQLType(typeof(AnyType))]
    public Dictionary<string, string>? EmailAliases { get; set; }
    public SlackWebHooksSettings? Webhooks { get; set; }
    public bool? NotificationsEnabled { get; set; }
}
